nrf_modem: doc: Update GNSS 1PPS documentation (#1090)
nrfconnect/sdk-nrfxlib: nrf_modem/doc/gnss_interface.rst

.. note::

>    1PPS feature must not be used when LTE is enabled.

Reviewer: ```suggestion
   Do not use the 1PPS feature when LTE is enabled.
```

Reviewer: I would have to disagree. In fact, "Do not use..." is actually a lot more aggressive way of putting this, but up to you. The original is not misleading by any means.

Author: Maybe it's just my background with IETF RFCs, where "must not" means "absolute prohibition" (https://datatracker.ietf.org/doc/html/rfc2119). 😄 

Ok, let's go with your suggestion.
